### PR TITLE
Jitter Review

### PR DESCRIPTION
**Pull request by rohbotics**

### CODE REVIEW

File: aruco_detect/src/map.cpp [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+    tf2::Vector3 o2 = t2.getOrigin();
+ 
+    t1.setOrigin((var1 * o2 + var2 * o1) / (var1 + var2));
+    
+    tf2::Quaternion q1 = t1.getRotation();
+    tf2::Quaternion q2 = t2.getRotation();
+    t1.setRotation(q1.slerp(q2, var1 / (var1 + var2)).normalize());
+}
+
+Observation::Observation(int fid, Vec3d &rvec, Vec3d &tvec, double ierr, double oerr) {
```
Comment (rohbotics) [inline aruco_detect/src/map.cpp]: Check that the rvec and tvec are handled appropriately. (Same coordinate systems, deg vs rad, etc)

File: aruco_detect/src/map.cpp [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+}
+
+Fiducial::Fiducial(int id, tf2::Transform &pose, double variance) {
+    this->id = id;
+    this->pose = pose;
+    this->variance = variance;
+    this->lastPublished = ros::Time(0);
+}
+
+Fiducial::Fiducial(int id, Vec3d rvec, Vec3d tvec, double variance) {
```
Comment (rohbotics) [inline aruco_detect/src/map.cpp]: Do we ever actually use this overload of the constructor? I don't see it any where.

Comment (rohbotics) [reply aruco_detect/src/map.cpp]: Also use refs

File: aruco_detect/src/map.cpp [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+                continue;
+            }
+
+            // dest is origin
+            if (fiducials.find(o2.fid) != fiducials.end() &&
+                fiducials[o2.fid].variance == 0) {
+                continue;
+            }
+
+            tf2::Transform T = fiducials[o1.fid].pose * o1.TfidCam * o2.TcamFid;
```
Comment (rohbotics) [inline aruco_detect/src/map.cpp]: Is the order of this multiplication correct? o1 vs o2?

File: aruco_detect/src/map.cpp [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+    ct.setOrigin(tf2::Vector3(
+       cameraTransform.transform.translation.x,
+       cameraTransform.transform.translation.y,
+       cameraTransform.transform.translation.z));
+ 
+    ct.setRotation(tf2::Quaternion(
+       cameraTransform.transform.rotation.x,
+       cameraTransform.transform.rotation.y,
+       cameraTransform.transform.rotation.z,
+       cameraTransform.transform.rotation.z));
```
Comment (rohbotics) [inline aruco_detect/src/map.cpp]: Bug! Should be cameraTransform.transform.rotation.w

File: aruco_detect/src/map.cpp [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+
+static void updateTransform(tf2::Transform &t1, double var1, tf2::Transform &t2, double var2) {
+    tf2::Vector3 o1 = t1.getOrigin();
+    tf2::Vector3 o2 = t2.getOrigin();
+ 
+    t1.setOrigin((var1 * o2 + var2 * o1) / (var1 + var2));
+    
+    tf2::Quaternion q1 = t1.getRotation();
+    tf2::Quaternion q2 = t2.getRotation();
+    t1.setRotation(q1.slerp(q2, var1 / (var1 + var2)).normalize());
```
Comment (rohbotics) [inline aruco_detect/src/map.cpp:90]: Should the variances be the same? David says that it doesn't really make sense, me might need to do multivariate stuff.

**Comment by jim-v** [comment]
Canceling this PR, since it is superseded by #58

**Pull request closed without merging**